packages/file_selector/file_selector_platform_interface/test/x_type_group_test.dart

```diff
@@ -71,7 +70,70 @@ void main() {
       expect(webOnly.allowsAny, false);
     });
 
-    test('Leading dots are removed from extensions', () {
+    test('passing only macUTIs should fill uniformTypeIdentifiers', () {
+      const List<String> macUTIs = <String>['public.plain-text'];
+      const XTypeGroup group = XTypeGroup(
+        macUTIs: macUTIs,
+      );
+
+      expect(group.uniformTypeIdentifiers, macUTIs);
+    });
+
+    test(
+        'passing only uniformTypeIdentifiers should fill uniformTypeIdentifiers',
+        () {
+      const List<String> uniformTypeIdentifiers = <String>['public.plain-text'];
+      const XTypeGroup group = XTypeGroup(
+        uniformTypeIdentifiers: uniformTypeIdentifiers,
+      );
+
+      expect(group.uniformTypeIdentifiers, uniformTypeIdentifiers);
+    });
+
+    test('macUTIs getter return macUTIs value passed in constructor', () {
+      const List<String> macUTIs = <String>['public.plain-text'];
+      const XTypeGroup group = XTypeGroup(
+        macUTIs: macUTIs,
+      );
+
+      expect(group.macUTIs, macUTIs);
+    });
+
+    test(
+        'macUTIs getter returns uniformTypeIdentifiers value passed in constructor',
+        () {
+      const List<String> uniformTypeIdentifiers = <String>['public.plain-text'];
+      const XTypeGroup group = XTypeGroup(
+        uniformTypeIdentifiers: uniformTypeIdentifiers,
+      );
+
+      expect(group.macUTIs, uniformTypeIdentifiers);
+    });
+
+    test('passing both uniformTypeIdentifiers and macUTIs should throw', () {
+      const List<String> macUTIs = <String>['public.plain-text'];
+      const List<String> uniformTypeIndentifiers = <String>[
+        'public.plain-images'
+      ];
+      expect(
+          () => XTypeGroup(
+              macUTIs: macUTIs,
+              uniformTypeIdentifiers: uniformTypeIndentifiers),
+          throwsA(predicate((Object? e) =>
+              e is AssertionError &&
+              e.message ==
+                  'Only one of uniformTypeIdentifiers or macUTIs can be non-null')));
```

Comment:
I'm not a fan of asserting the text of the assertion error, I'd just do a `expect(closure, throwsAssertionError)` (or `throwsA(isA<AssertionError>())`)